thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/datasource/pinot/SqlUtils.java

```diff
@@ -189,9 +195,13 @@ public static String getDimensionAsMetricSql(ThirdEyeRequest request, MetricFunc
         request.getGroupBy(), request.getGroupByTimeGranularity(), dataTimeSpec,
         metricNamesList, metricNamesColumnsList, metricValuesColumn, request.getLimit());
 
-    return dimensionAsMetricPql;
+    return escapeSqlReservedKeywords(dimensionAsMetricPql);
   }
 
+  private static String escapeSqlReservedKeywords(String query) {
+    // escape all reserve keywords with double quotes
+    return RESERVED_KEYWORD_PATTERN.matcher(query).replaceAll(RESERVED_KEYWORD_REPLACEMENT);
```

Comment:
Rather than maintaining a reserved keyword list, it might be better to escape every dimension?

Reply:
Actually, it is pretty standard to escape the keyword only. Adopted this code from a library.